### PR TITLE
docs: add DatePicker format demo(Week start ~ end)

### PR DESCRIPTION
**Pull request by Rain120**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |     周选择器没有自定义渲染的demo，新增 DatePicker format demo     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/date-picker/demo/format.md [hunk 1]
```diff
@@ -20,17 +20,23 @@ import moment from 'moment';
...
+  `${moment(value).startOf('week').format(weekFormat)} ~ ${moment(value)
+    .endOf('week')
+    .format(weekFormat)}`;
+
 ReactDOM.render(
   <Space direction="vertical" size={12}>
     <DatePicker defaultValue={moment('2015/01/01', dateFormat)} format={dateFormat} />
     <DatePicker defaultValue={moment('01/01/2015', dateFormatList[0])} format={dateFormatList} />
-    <DatePicker defaultValue={moment('2015/01', monthFormat)} format={monthFormat} picker="month" />
+    <DatePicker defaultValue={moment()} format={customWeekStartEndFormat} picker="week" />
```
Comment (xrkffgg) [inline components/date-picker/demo/format.md:39]: 月份不用删呀，新增 week

CI 挂了

Comment (Rain120) [reply components/date-picker/demo/format.md:39]: 1. 月份确实是我误删了
2. CI 这个问题，我只改了doc，应该是不影响的，为什么会挂啊

Comment (xrkffgg) [reply components/date-picker/demo/format.md:39]: 可以点开看详情。 demo 是有测试的

Comment (Rain120) [reply components/date-picker/demo/format.md:40]: 好的

**Comment by xrkffgg** [comment]
执行下 yarn test -u

**Comment by Rain120** [comment]
我尝试着单独测试我这个组件，发现都是可以跑通的，但是有一些旧版的废弃提示

**Comment by xrkffgg** [comment]
更新下 snap 一起提上来~

**Comment by Rain120** [comment]
已提交

**Comment by afc163** [comment]
Rebase master please

**Comment by Rain120** [comment]
Done

**Comment by xrkffgg** [comment]
> Done

push 的时候要 force `git push -f origin xx:xx`。也可以参照 https://github.com/ant-design/ant-design/issues/30441

要不你重新开个 PR 吧。

**Comment by Rain120** [comment]
好的

**Comment by Rain120** [comment]
新PR [here](https://github.com/ant-design/ant-design/pull/32078)

**Pull request closed without merging**